extensions/applicationinsights-properties-js/Tests/Unit/src/properties.tests.ts

```diff
@@ -375,8 +372,229 @@ export class PropertiesTests extends AITestClass {
+             
+                let os = propCtx.os;
+                Assert.equal(os, null, "os should be null");
+            
+                let web = propCtx.web;
+                Assert.equal(web, null, "web should be null");
+
+                let sessionId = propCtx.getSessionId();
+                Assert.equal(sessionId, null, "session Id should be null");
+                
+                // change properities here to make sure we won't overwrite them after config change
+                device.id = deviceId;
+                location.ip = locId;
+                trace.parentID = parentId;
+                application.build = appBuild;
+                session.id = seId;
+                sessionMgr.automaticSession.id = seId;
+                Assert.equal(propCtx.getSessionId(), seId, "session Id should be updated test1");
+                let sessionPrefix = "ai_session" + "postfix1";
+                sessionMgr.backup();
+                let sessionStorage = utlGetLocalStorage(core.logger, sessionPrefix);
+                Assert.ok(sessionStorage.indexOf(seId) > -1, "sessionStorage should be set based on session id test1");
+
+                sessionMgr.update();
+                Assert.ok(cookieStub.called, "cookie set test1");
+                Assert.equal(sessionPrefix, cookie, "cookie name is set test1");
+                Assert.ok(cookieValue.indexOf(seId) > -1, "cookie value is set test1");
+                this.clock.tick(20);
+                sessionMgr.update();
+                Assert.equal(sessionMgr.automaticSession.id, seId, "session id should be same test1");
+                this.clock.tick(24 * 60 * 60 * 101 - 20);
+                sessionMgr.update();
+                Assert.equal(sessionMgr.automaticSession.id, "26", "session id should be renewed test1");
+                Assert.ok(cookieValue.indexOf("26|") > -1,"cookie value is reset test1");
+              
+                // change config
+                let newConfig = {
+                    accountId: "id2",
+                    sessionRenewalMs: 30 * 60 * 102,
+                    samplingPercentage: 90,
+                    sessionExpirationMs: 24 * 60 * 60 * 102,
+                    cookieDomain: "domain2",
+                    sdkExtension: "ext2",
+                    isBrowserLinkTrackingEnabled: true,
+                    appId: "id2",
+                    getSessionId: "session2",
+                    namePrefix: "prefix2",
+                    sessionCookiePostfix: "postfix2",
+                    userCookiePostfix: "usercookie2",
+                    idLength: 26,
+                    getNewId: (idLength?: number) => {
+                        return "" + (idLength || 0) + 1;
+                    }
+                } as IPropertiesConfig;
+
+                core.config.extensionConfig =  core.config.extensionConfig?  core.config.extensionConfig : {};
+                core.config.extensionConfig[id] = newConfig;
```

Comment:
This is the "Key" change from the previous test, where this will cause the properties of `newConfig` to be copied to `core.config.extensionConfig[id]` rather than getting directly assigned (and overwriting) the previous value.
This occurs because the `extensionConfig` and its properties are marked as referenced (using the new `ref()`) as part of the dynamic configuration.

This is required because the PropertiesPlugin passes down it's configuration to its supporting components, and if this allowed the config to be "replaced" those support components would never be notified or receive the "updated/changed" configuration values (because they would be a different object instance)